simulariumio/converter.py

```diff
@@ -279,6 +280,35 @@ def _determine_plot_reader(plot_type: str = "scatter") -> [Reader]:
 
         raise UnsupportedPlotTypeError(plot_type)
 
+    def _check_agent_ids_are_unique_per_frame(self) -> bool:
+        """
+        For each frame, check that none of the unique agent IDs overlap
+        """
+        bundleData = self._data["spatialData"]["bundleData"]
+        for t in range(len(bundleData)):
+            data = bundleData[t]["data"]
+            next_uid_index = 1
+            uids = []
+            get_n_subpoints = False
+            for i in range(len(data)):
+                if i == next_uid_index:
```

Comment:
This block could be accomplished in one loop - not strictly needed but easier to read at a glance imo:

e.g. while(i < len(data)) {
... assign directly to i instead of next_uid_index;
}